src/Microsoft.SqlTools.ServiceLayer/Workspace/Contracts/TextDocument.cs

```diff
@@ -156,7 +156,36 @@ public class Position
         /// <summary>
         /// Gets or sets the zero-based column number.
         /// </summary>
-        public int Character { get; set; }
+        public int Character { get; set; }
+
+        /// <summary>
+        /// Overrides the base equality method
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
+        public override bool Equals(object obj)
+        {
+            if (obj == null || GetType() != obj.GetType())

```

Comment:
What if someone has a subclass of position? I think IsInstanceOf or similar (we have cases of this in the project) work better. You could also just do obj as Position and check if null